pkg/tfshim/sdk-v2/provider2.go

```diff
@@ -637,11 +637,26 @@ func (s *grpcServer) ApplyResourceChange(
 	if err != nil {
 		return nil, err
 	}
+
+	var providerMetaVal []byte
+	if providerMeta != nil {
+		providerMetaVal, err = msgpack.Marshal(*providerMeta, providerMeta.Type())
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		providerMetaVal, err = msgpack.Marshal(cty.NullVal(cty.EmptyObject), cty.EmptyObject)
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
The previous compare function wasn't catching this, but the new comparison does. This now more accurately imitates TF.